proto/pulumi/provider.proto

```diff
@@ -133,11 +133,20 @@ service ResourceProvider {
     // difference (if any) between them. `Diff` should only be called with values that have at some point been validated
     // by a [](pulumirpc.ResourceProvider.Check) call.
     rpc Diff(DiffRequest) returns (DiffResponse) {}
-    // Create allocates a new instance of the provided resource and returns its unique ID afterwards.  (The input ID
-    // must be blank.)  If this call fails, the resource must not have been created (i.e., it is "transactional").
+
+    // `Create` provisions a new instance of the specified [(custom) resource](custom-resources). It returns a
+    // provider-assigned ID for the resource as well as the output properties that arose from the creation properties.
+    // Output properties are typically the union of the resource's input properties and any additional values that were
+    // computed or made available during creation.
+    //
+    // If creation fails, `Create` may return an [](pulumirpc.ErrorResourceInitFailed) error detail explaining why.
```

Comment:
An idempotency discussion might be pertinent here, but maybe later as we have not figured it out. If a provider can signal that a resource is "already created" how can it do that.

There is definitely a partially created state resources enter, I think with pulumirpc.ErrorResourceInitFailed, they get tracked in state but it is still considered an error. I'm confused by that and would appreciate a page explaining that concept specifically, probably just to link here.

Reply:
Tracked in #17691